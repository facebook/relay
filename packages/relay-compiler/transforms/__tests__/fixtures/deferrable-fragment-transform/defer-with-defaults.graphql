query TestDeferWithVariableWithDefaultValue($size: [Int]) {
  me {
    ...ProfilePic @relay(deferrable: true)
  }
}

fragment ProfilePic on User @argumentDefinitions(
  size: {type: "[Int]", defaultValue: [100]}
) {
  profilePicture(size: $size) {
    uri
    width
    height
  }
}
