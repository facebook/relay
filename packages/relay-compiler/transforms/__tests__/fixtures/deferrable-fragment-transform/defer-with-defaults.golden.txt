query TestDeferWithVariableWithDefaultValue(
  $size: [Int]
) {
  me {
    ... on User {
      ... on Node {
        ProfilePic_deferredID: id
      }
    }
  }
}

fragment ProfilePic on User @argumentDefinitions(
  size: {type: "[Int]", defaultValue: [100]}
) {
  profilePicture(size: $size) {
    uri
    width
    height
  }
}

query ProfilePic_Deferred(
  $deferredID: ID
  $size: [Int] = [100]
) {
  node(id: $deferredID) {
    ...ProfilePic
  }
}


{
  "kind": "Root",
  "operation": "query",
  "metadata": null,
  "name": "TestDeferWithVariableWithDefaultValue",
  "dependentRequests": [
    {
      "operationName": "ProfilePic_Deferred",
      "argumentDependencies": [
        {
          "kind": "ArgumentDependency",
          "argumentName": "deferredID",
          "fromName": "TestDeferWithVariableWithDefaultValue",
          "fromPath": "me.ProfilePic_deferredID",
          "ifList": "each",
          "ifNull": "skip"
        }
      ]
    }
  ],
  "argumentDefinitions": [
    {
      "kind": "LocalArgumentDefinition",
      "metadata": null,
      "name": "size",
      "defaultValue": null,
      "type": "[Int]"
    }
  ],
  "directives": [],
  "selections": [
    {
      "kind": "LinkedField",
      "alias": null,
      "args": [],
      "directives": [],
      "metadata": null,
      "name": "me",
      "selections": [
        {
          "kind": "InlineFragment",
          "metadata": null,
          "typeCondition": "User",
          "directives": [],
          "selections": [
            {
              "kind": "InlineFragment",
              "metadata": null,
              "typeCondition": "Node",
              "directives": [],
              "selections": [
                {
                  "kind": "ScalarField",
                  "name": "id",
                  "alias": "ProfilePic_deferredID",
                  "args": [],
                  "handles": null,
                  "directives": [],
                  "metadata": {
                    "deferrable": true
                  },
                  "type": "ID"
                }
              ]
            }
          ]
        }
      ],
      "type": "User"
    }
  ],
  "type": "Query"
}
{
  "kind": "Fragment",
  "directives": [],
  "metadata": null,
  "name": "ProfilePic",
  "selections": [
    {
      "kind": "LinkedField",
      "alias": null,
      "args": [
        {
          "kind": "Argument",
          "metadata": null,
          "name": "size",
          "value": {
            "kind": "Variable",
            "metadata": null,
            "variableName": "size",
            "type": "[Int]"
          },
          "type": "[Int]"
        }
      ],
      "directives": [],
      "metadata": null,
      "name": "profilePicture",
      "selections": [
        {
          "kind": "ScalarField",
          "alias": null,
          "args": [],
          "directives": [],
          "metadata": null,
          "name": "uri",
          "type": "String"
        },
        {
          "kind": "ScalarField",
          "alias": null,
          "args": [],
          "directives": [],
          "metadata": null,
          "name": "width",
          "type": "Int"
        },
        {
          "kind": "ScalarField",
          "alias": null,
          "args": [],
          "directives": [],
          "metadata": null,
          "name": "height",
          "type": "Int"
        }
      ],
      "type": "Image"
    }
  ],
  "type": "User",
  "argumentDefinitions": [
    {
      "kind": "LocalArgumentDefinition",
      "defaultValue": [
        100
      ],
      "metadata": null,
      "name": "size",
      "type": "[Int]"
    }
  ]
}
{
  "kind": "Root",
  "operation": "query",
  "metadata": null,
  "name": "ProfilePic_Deferred",
  "dependentRequests": [],
  "argumentDefinitions": [
    {
      "kind": "LocalArgumentDefinition",
      "metadata": null,
      "name": "deferredID",
      "defaultValue": null,
      "type": "ID"
    },
    {
      "kind": "LocalArgumentDefinition",
      "metadata": null,
      "name": "size",
      "type": "[Int]",
      "defaultValue": [
        100
      ]
    }
  ],
  "directives": [],
  "selections": [
    {
      "kind": "LinkedField",
      "name": "node",
      "alias": null,
      "args": [
        {
          "kind": "Argument",
          "name": "id",
          "metadata": null,
          "value": {
            "kind": "Variable",
            "variableName": "deferredID",
            "metadata": null,
            "type": "ID"
          },
          "type": "ID"
        }
      ],
      "directives": [],
      "metadata": null,
      "handles": null,
      "selections": [
        {
          "kind": "FragmentSpread",
          "args": [],
          "metadata": null,
          "name": "ProfilePic",
          "directives": []
        }
      ],
      "type": "Node"
    }
  ],
  "type": "Query"
}
